src/test/groovy/com/synacy/gradprogram/spock/exercise/OrderingServiceSpec.groovy

```diff
@@ -0,0 +1,171 @@
+    }
+
+    def "test if the Items will be stored, return true if all the items were stored"(){
+     given:
+        def items = [
+                new Item("Charger", 100.00, ItemType.GADGET),
+                new Item("Vanz Tshirt", 160.00, ItemType.CLOTHING),
+                new Item("Oven", 800.00, ItemType.APPLIANCE),
+                new Item("555 Tuna", 25.00, ItemType.FOOD),
+                new Item("Pork Chop", 150.00, ItemType.FOOD)
+        ]
+        when:
+        boolean ifcontainsFoodType = orderingService.cartContainsFoodItem(cart = new Cart(UUID.randomUUID(), items))
+
+        then:
+        ifcontainsFoodType
+    }
+
+    def "Should Determine the Total Cost of the Cart is true"(){
+        given:
+        def totalItems = [
+                new Item("555 Tuna", 50.00, ItemType.FOOD),
+                new Item("Pork Chop", 150.00, ItemType.FOOD),
+                new Item("Charger", 100.00, ItemType.GADGET),
+                new Item("Vanz Tshirt", 160.00, ItemType.CLOTHING),
+                new Item("Oven", 800.00, ItemType.APPLIANCE)
+        ]
+        when:
+        double totalPrice = orderingService.calculateTotalCostOfCart(cart = new Cart(UUID.randomUUID(),totalItems))
+
+        then:
+        println(totalPrice)
+        totalPrice == 1260.00
+
+    }
+
+
+    def "Should return the items is Eligible for a Discount"(){
+        given:
+        def items = [
+                new Item("555 Tuna", 50.00, ItemType.FOOD),
+                new Item("Pork Chop", 150.00, ItemType.FOOD),
+                new Item("Charger", 100.00, ItemType.GADGET),
+                new Item("Vanz Tshirt", 160.00, ItemType.CLOTHING),
+                new Item("Oven", 800.00, ItemType.APPLIANCE),
+                new Item("Fan", 200.00, ItemType.APPLIANCE),
+        ]
+        when:
+        boolean eligibleForDiscount = orderingService.isCartEligibleForDiscount(cart = new Cart(UUID.randomUUID(),items))
+
+
+        then:
+        println "Total Price of Cart: ${orderingService.calculateTotalCostOfCart(cart)}"
+        println "Number of Items in Cart: ${cart.getItems().size()}"
+        eligibleForDiscount
+    }
+
+    def "Apply discount if the Cart is Eligble for discount"(){
+        given:
+        def items = [
+                new Item("555 Tuna", 50.00, ItemType.FOOD),
+                new Item("Pork Chop", 150.00, ItemType.FOOD),
+                new Item("Charger", 100.00, ItemType.GADGET),
+                new Item("Vanz Tshirt", 160.00, ItemType.CLOTHING),
+                new Item("Oven", 800.00, ItemType.APPLIANCE),
+                new Item("Fan", 200.00, ItemType.APPLIANCE),
+        ]
+        when:
+        double originalTotalCost = orderingService.calculateTotalCostOfCart(cart = new Cart(UUID.randomUUID(), items))
+        orderingService.applyDiscountToCartItems(cart)
+
+        then:
+        orderingService.isCartEligibleForDiscount(cart) == true
+        double discountedTotalCost = (originalTotalCost * 0.90)
```

Comment:
Let's try to keep it simple, unit tests shouldn't contain any logic and call any other service method aside from the one we are testing.